streams/src/test/java/org/apache/kafka/streams/kstream/internals/TimeWindowedCogroupedKStreamImplTest.java

```diff
@@ -0,0 +1,264 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.Properties;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.TestInputTopic;
+import org.apache.kafka.streams.TopologyTestDriver;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.Grouped;
+import org.apache.kafka.streams.kstream.CogroupedKStream;
+import org.apache.kafka.streams.kstream.KGroupedStream;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.kstream.Named;
+import org.apache.kafka.streams.kstream.TimeWindowedCogroupedKStream;
+import org.apache.kafka.streams.kstream.TimeWindows;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.state.ValueAndTimestamp;
+import org.apache.kafka.streams.state.WindowStore;
+import org.apache.kafka.test.MockAggregator;
+import org.apache.kafka.test.MockInitializer;
+import org.apache.kafka.test.MockProcessorSupplier;
+import org.apache.kafka.test.StreamsTestUtils;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class TimeWindowedCogroupedKStreamImplTest {
+
+    private final MockProcessorSupplier<Windowed<String>, String> processorSupplier = new MockProcessorSupplier<>();
+    private static final String TOPIC = "topic";
+    private static final String TOPIC2 = "topic2";
+    private final StreamsBuilder builder = new StreamsBuilder();
+    private KGroupedStream<String, String> groupedStream;
+
+    private KGroupedStream<String, String> groupedStream2;
+    private CogroupedKStream<String, String> cogroupedStream;
+    private TimeWindowedCogroupedKStream<String, String> windowedCogroupedStream;
+
+    private final Properties props = StreamsTestUtils
+        .getStreamsConfig(Serdes.String(), Serdes.String());
+
+    @Before
+    public void setup() {
+        final KStream<String, String> stream = builder.stream(TOPIC, Consumed
+            .with(Serdes.String(), Serdes.String()));
+        final KStream<String, String> stream2 = builder.stream(TOPIC2, Consumed
+            .with(Serdes.String(), Serdes.String()));
+
+        groupedStream = stream.groupByKey(Grouped.with(Serdes.String(), Serdes.String()));
+        groupedStream2 = stream2.groupByKey(Grouped.with(Serdes.String(), Serdes.String()));
+        cogroupedStream = groupedStream.cogroup(MockAggregator.TOSTRING_ADDER).cogroup(groupedStream2, MockAggregator.TOSTRING_REMOVER);
+        windowedCogroupedStream = cogroupedStream.windowedBy(TimeWindows.of(ofMillis(500L)));
+    }
+
+    @Test
+    public void timeWindowTest() {
+        assertNotNull(windowedCogroupedStream);
```

Comment:
What value does this test provide? Seems it verifies that `setup()` method is correct? Seems unnecessary to me.